Header header        # Header timestamp should be acquisition time of image
                     # Header frame_id should be optical frame of camera
                     # origin of frame should be optical center of camera
                     # +x should point to the right in the image
                     # +y should point down in the image
                     # +z should point into to plane of the image
                     # If the frame_id here and the frame_id of the CameraInfo
                     # message associated with the image conflict
                     # the behavior is undefined
KeyPoints_vector    keypoints
Descriptors_vector  descriptors               
sensor_msgs/Image   depth 
float64[12]         P     
sensor_msgs/Image   image
